inc/javascript.inc.php

```diff
@@ -357,26 +324,31 @@ function calculateFromForeignAmount(element, in_flag) {
  * set the other to 0.
  * Used in journal/edit view.
  */
-function allowOnlyCreditOrDebit(element, credit_or_debit) {
-  var in_amount    = element.parentElement.parentElement.getElementsByTagName('input')[0];
-  var out_amount   = element.parentElement.parentElement.getElementsByTagName('input')[1];
-  var conv_rate    = document.getElementsByName('voucher.ForeignConvRate')[0];
-  var currency     = document.getElementsByName('voucher.ForeignCurrencyID')[0];
-  var f_in_amount  = element.parentElement.parentElement.getElementsByTagName('input')[2];
-  var f_out_amount = element.parentElement.parentElement.getElementsByTagName('input')[3];
+function allowOnlyCreditOrDebit(element, credit_or_debit, from_currency) {
+  if(from_currency) {

```

Comment:
Investigate if we need `<div class='currency_fields'>`. If not, simplify: add the class to the inputs.